runtime/cudaq/algorithms/observe.h

```diff
@@ -75,14 +80,17 @@ std::optional<observe_result>
 runObservation(KernelFunctor &&k, cudaq::spin_op &h, quantum_platform &platform,
                int shots, const std::string &kernelName, std::size_t qpu_id = 0,
                details::future *futureResult = nullptr,
-               std::size_t batchIteration = 0,
-               std::size_t totalBatchIters = 0) {
+               std::size_t batchIteration = 0, std::size_t totalBatchIters = 0,
+               std::optional<std::size_t> numTrajectories = {}) {
   auto ctx = std::make_unique<ExecutionContext>("observe", shots);
   ctx->kernelName = kernelName;
   ctx->spin = &h;
   if (shots > 0)
     ctx->shots = shots;
 
+  if (numTrajectories.has_value())
+    ctx->numberTrajectories = numTrajectories.value();
```

Comment:
```suggestion
    ctx->numberTrajectories = *numTrajectories;
```
Should be safe (checked in previous line) and foregoes a second check and pessimistically thinking an exception could be thrown.